google/cloud/aiplatform/datasets/column_names_dataset.py

```diff
@@ -0,0 +1,344 @@
+                field
+            )
+        }
+
+        # Only return "leaf nodes", basically any field that doesn't have children
+        if len(ancestor_names) == 0:
+            return {schema_field.name}
+        else:
+            return {f"{schema_field.name}.{name}" for name in ancestor_names}
+
+    @staticmethod
+    def _retrieve_bq_source_columns(
+        project: str,
+        bq_table_uri: str,
+        credentials: Optional[auth_credentials.Credentials] = None,
+    ) -> Set[str]:
+        """Retrieve the column names from a table on Google BigQuery
+        Nested schema fields are flattened and concatenated with a ".".
+        Schema fields with child fields are not included, but the children are.
+
+        Example Usage:
+
+            column_names = _retrieve_bq_source_columns(
+                "project_id",
+                "bq://project_id.dataset.table"
+            )
+
+            # column_names = {"column_1", "column_2", "column_3.nested_field"}
+
+        Args:
+            project (str):
+                Required. Project to initiate the BigQuery client with.
+            bq_table_uri (str):
+                Required. A URI to a BigQuery table.
+                Can include "bq://" prefix but not required.
+            credentials (auth_credentials.Credentials):
+                Credentials to use with BQ Client.
+
+        Returns:
+            Set[str]
+                A set of column names in the BigQuery table.
+        """
+
+        # Remove bq:// prefix
+        prefix = "bq://"
+        if bq_table_uri.startswith(prefix):
+            bq_table_uri = bq_table_uri[len(prefix) :]
+
+        client = bigquery.Client(project=project, credentials=credentials)
+        table = client.get_table(bq_table_uri)
+        schema = table.schema
+
+        return {
+            field_name
+            for field in schema
+            for field_name in _ColumnNamesDataset._get_bq_schema_field_names_recursively(
+                field
+            )
+        }
+
+    def _get_default_column_transformations(
+        self, target_column: str,
+    ) -> Tuple[Dict, List[str]]:
+        """Get default column transformations from the column names, while omitting the target column.
+
+        Args:
+            target_column (str):
+                Required. The name of the column values of which the Model is to predict. 
+
+        Returns:
+            Dict
+                The default column transformations.
+        """
+
+        column_names = [
+            column_name
+            for column_name in self.column_names
+            if column_name != target_column
+        ]
+        column_transformations = [
+            {"auto": {"column_name": column_name}} for column_name in column_names
+        ]
+
+        return (column_transformations, column_names)
+
+    @staticmethod
+    def _validate_and_get_column_transformations(
```

Comment:
It seems like this method should be a part of the training job.

Reply:
Agreed, will create an abstraction and move it back.